openmls/src/group/mls_group/processing.rs

```diff
@@ -141,18 +146,31 @@ impl MlsGroup {
         self.own_kpbs.clear();
 
         // Delete a potential pending commit
-        self.clear_pending_commit();
+        self.clear_pending_commit()?;
 
         Ok(())
     }
 
     /// Merges the pending [`StagedCommit`] and, if the merge was successful,
     /// clears the field by setting it to `None`.
     pub fn merge_pending_commit(&mut self) -> Result<(), MlsGroupError> {
-        if let Some(staged_commit) = self.pending_commit.take() {
-            self.merge_staged_commit(staged_commit)
-        } else {
-            Err(MlsGroupError::NoPendingCommit)
+        match &self.group_state {
+            MlsGroupState::PendingCommit(_) => {
+                let old_state = mem::replace(&mut self.group_state, MlsGroupState::Operational);
+                if let MlsGroupState::PendingCommit(pending_commit_state) = old_state {
+                    self.merge_staged_commit(pending_commit_state.into())
+                } else {
+                    // We should never reach this state, as we previously
+                    // checked that there actually is a pending commit.
+                    Err(MlsGroupError::LibraryError(
+                        "Reached unreachable state during pending commit merge.".into(),
```

Comment:
Why do we need the `else` clause at all?

Reply:
Because the `match` needs to return something if `old_state` is not a `PendingCommit` (which it can't be, hence the error). Is that what you meant?

Reply:
Since the function returns `()` you don't have to return something here. You can just return `Ok(())` in the end and abort in line 161 if there's an error and then delete the whole `else` clause since we know for sure what's happening.

Reply:
I see what you mean. That does indeed sound a bit cleaner.